### PR TITLE
improvement in the tf.nn.raw_rnn documentation

### PR DESCRIPTION
**Pull request by akanimax**
PR for #14963. This exposes the advanced use-case of the raw_rnn where the user can project the cell_output into logits as a part of the `loop_fn` computations. Earlier it mislead the users that it is mandatory for the `loop_fn` to return `emit_output` such that it matches the `cell.output_size`.

### CODE REVIEW

**Comment by akanimax** [comment]
I have signed the CLA!

**Comment by akanimax** [comment]
I have signed the CLA again. could you please check?

**Comment by drpngx** [comment]
@ebrevdo WDYT?

**Comment by ebrevdo** [comment]
This is more than a documentation improvement. There are code changes here.

**Comment by ebrevdo** [comment]
Update the pr title

**Comment by akanimax** [comment]
@ebrevdo there are no code changes in the PR. There are changes in the pseudocode given in the documentation of tf.nn.raw_rnn

**Comment by ebrevdo** [comment]
Thanks for clarifying; you're right.  The github context window was too small :(

**Review by ebrevdo — APPROVED** [review]

**Comment by drpngx** [comment]
```
FAIL: Found 5 non-whitelited pylint errors:
tensorflow/python/ops/rnn.py:946: [C0301(line-too-long), ] Line too long (84/80)
tensorflow/python/ops/rnn.py:947: [C0301(line-too-long), ] Line too long (92/80)
tensorflow/python/ops/rnn.py:948: [C0301(line-too-long), ] Line too long (85/80)
tensorflow/python/ops/rnn.py:949: [C0301(line-too-long), ] Line too long (86/80)
tensorflow/python/ops/rnn.py:950: [C0301(line-too-long), ] Line too long (91/80)
```

**Comment by akanimax** [comment]
@drpngx The pylint errors are on lines 946 ~ 950 where I have made no changes. Is it okay to bring them within the horizontal range of 80 too?

**Comment by akanimax** [comment]
@drpngx the lines were exceeding the 80 chars limit. I have fixed it in the subsequent commit.

**Comment by drpngx** [comment]
Thanks, testing again.

**Comment by drpngx** [comment]
Yay!

**Comment by akanimax** [comment]
Thank you for adding my contributions to the repo. :+1: 
Cheers!
